tests/page-objects/default/enketo/common-enketo.wdio.page.js

```diff
@@ -0,0 +1,74 @@
+const currentSection =  () => $('section[class*="current"]');
+
+const divContainer = () => $('div.container');
+
+const getCurrentPageSection = async () => await currentSection().isExisting() ? currentSection() : divContainer();
+
+const enabledFieldset = (section) => section.$$('fieldset.or-branch:not(.disabled)');
+
+const getCorrectFieldsetSection = async (section) => {
+  if (await enabledFieldset(section).length === 1){
+    return enabledFieldset(section)[0];
+  }
+  return section;
+};
+
+const isElementDisplayed = async (type, text) => {
+  return await (await getCurrentPageSection()).$(`${type}*=${text}`).isDisplayed();
+};
+
+const selectRadioButton = async (question, label) => {
+  const radioButton = await (await getCurrentPageSection())
+    .$(`legend*=${question}`)
+    .parentElement()
+    .$(`label*=${label}`);
+  await radioButton.waitForClickable();
+  await radioButton.click();
+};
+
+const selectCheckBox = async (question, text) => {
+  const page = await getCurrentPageSection();
+  const checkbox = await (await getCorrectFieldsetSection(page))
+    .$(`legend*=${question}`)
+    .nextElement()
+    .$(`label*=${text}`);
+  await checkbox.waitForClickable();
+  await checkbox.click();
+};
+
+const setValue = async (typeSelector, question, value) => {
+  const element = await (await getCurrentPageSection())
+    .$(`label*=${question}`)
+    .$(typeSelector);
+  await element.waitForDisplayed();
+  await element.setValue(value);
+};
+
+const setInputValue = async (question, value) => {
+  await setValue('input', question, value);
+};
+
+const setDateValue = async (question, value) => {
+  await setValue('input.ignore.input-small', question, value);
+};
+
+const setTextareaValue = async (question, value) => {
+  await setValue('textarea', question, value);
+};
+
+const validateSummaryReport = async (textArray) => {
+  const element = await getCurrentPageSection();
+  for (const text of textArray) {
+    expect(await (await element.$(`span*=${text}`)).isDisplayed()).to.be.true;
+  }
+};
```

Comment:
I'm trying to figure out if we can do something more specific than just checking the page for the strings, for example, asserting the order or the format somehow. I can't think of anything though. The summary page isn't well structured. Can you think of anything?

One day it'd be great to take a snapshot of the page and compare build to build to make sure nothing changes but that day isn't today...

Reply:
I thought about that when I was making the changes, but most of the summary pages are different depending on the report, and this was the easiest way I found to verify it.

Of course, I am open to suggestions, if you or anyone else on the team have an idea of how can we make it better we can definitely try it 🙂